Objects/object.c

```diff
@@ -1364,6 +1364,12 @@ PyObject_GenericGetAttr(PyObject *obj, PyObject *name)
     return _PyObject_GenericGetAttrWithDict(obj, name, NULL, 0);
 }
 
+PyObject *
+PyObject_GenericTryGetAttr(PyObject *obj, PyObject *name)
+{
+    return _PyObject_GenericGetAttrWithDict(obj, name, NULL, 1);
+}
+
```

Comment:
Since this is private, we should prefix this with an underscore by convention.

```suggestion
PyObject *
_PyObject_GenericTryGetAttr(PyObject *obj, PyObject *name)
{
    return _PyObject_GenericGetAttrWithDict(obj, name, NULL, 1);
}

```

Also, could you add some comment above the function saying why it is different and when to use it?